mlir/include/mlir/Dialect/AMDGPU/IR/AMDGPU.td

```diff
@@ -898,6 +898,27 @@ def AMDGPU_GatherToLDSOp :
   let hasVerifier = 1;
 }
 
+def AMDGPU_TransposeLoadOp :
+    AMDGPU_Op<"transpose_load", [SameVariadicOperandSize]>,
+    Arguments<(ins Arg<AnyMemRef, "buffer to transpose load from", [MemRead]>:$src, Variadic<Index>:$srcIndices)>,
+    Results<(outs MFMAInTypes:$dst)> {
```

Comment:
I don't think we need to restrict to the MFMAInTypes - that'll give the wrong impression. We just load a vector, and then impose the condition that the total returned size of the vector is 64-bits and that your element type is {4, 6, 8, 16} bits long.

Reply:
I actually think the idea is good. I updated the code accordingly.